test/membrane_rtc_engine/webrtc/simulcast_tee_test.exs

```diff
@@ -7,128 +7,191 @@ defmodule Membrane.RTC.Engine.Endpoint.WebRTC.SimulcastTeeTest do
-      assert_sink_event(pipeline, {:source, "m"}, %Membrane.KeyframeRequestEvent{}, 10_000)
-
-      # FIXME
-      # This shouldn't be needed. There is a bug in forwarder.
-      # Consider scenario:
-      # h -> inactive h -> enque m -> active h
-      # We won't send keyframe when h becomes active again
-      # until we receive keyframe for m
-      flush_buffers(pipeline)
-      assert_sink_buffer(pipeline, :sink, _buffer)
-
-      actions = [event: {:output, %TrackVariantResumed{}}]
-      Pipeline.execute_actions(pipeline, forward: {{:source, "h"}, {:execute_actions, actions}})
-      Pipeline.execute_actions(pipeline, forward: {{:source, "h"}, {:set_active, true}})
-      assert_sink_event(pipeline, {:source, "h"}, %Membrane.KeyframeRequestEvent{}, 15_000)
-
-      Pipeline.terminate(pipeline, blocking?: true)
+    receive do
+      {:DOWN, _ref, :process, ^pipeline, {:shutdown, :child_crash}} -> :ok
     end
   end
 
-  defp build_pipeline() do
-    track =
-      Track.new(
-        :video,
-        @track_id,
-        "generated",
-        :H264,
-        90_000,
-        :raw,
-        nil,
-        id: @track_id,
-        simulcast_encodings: @layers
-      )
-
-    layers_links =
-      for layer <- @layers do
-        source = %TestSource{
-          interval: Time.milliseconds(50),
-          payload: fake_keyframe(layer)
-        }
-
-        link({:source, layer}, source)
-        |> to({:realtimer, layer}, Membrane.Realtimer)
-        |> via_in(Pad.ref(:input, {@track_id, layer}))
-        |> to(:tee)
-      end
+  test "SimulcastTee doesn't send data until receiving RequestTrackVariant event and a keyframe for requested track variant" do
+    track = build_h264_track()
+    pipeline = build_video_pipeline(track, {nil, &Utils.generator/2})
 
-    links = [
-      link(:tee, %SimulcastTee{track: track})
-      |> via_out(Pad.ref(:output, {:endpoint, @endpoint_id}))
-      |> to(:sink, Sink)
-      | layers_links
+    Enum.each(track.simulcast_encodings, fn encoding ->
+      assert_sink_event(pipeline, :sink, %TrackVariantResumed{variant: ^encoding})
+    end)
+
+    # tee shouldn't send us any packets
+    # until we request specific track variant
+    refute_sink_buffer(pipeline, :sink, _buffer)
+
+    actions = [event: {:input, %RequestTrackVariant{variant: "h"}}]
+    Pipeline.execute_actions(pipeline, forward: {:sink, {:execute_actions, actions}})
+    assert_sink_event(pipeline, {:source, "h"}, %Membrane.KeyframeRequestEvent{})
+
+    refute_sink_buffer(pipeline, :sink, _buffer)
+
+    actions = [
+      buffer: {:output, %Buffer{payload: <<>>, metadata: %{is_keyframe: true}}}
     ]
 
-    {:ok, pipeline} = Pipeline.start_link(links: links)
+    Pipeline.execute_actions(pipeline, forward: {{:source, "h"}, {:execute_actions, actions}})
 
-    assert_pipeline_playback_changed(pipeline, :prepared, :playing)
-    assert_start_of_stream(pipeline, :sink)
+    # TODO assert we receive TrackVariantSwitched before any buffer
```

Comment:
Welp, I don't really know how to achieve that :sweat_smile:

Reply:
Wouldn't `refute_sink_buffer` with timeout 0 work here?

Reply:
I guess we would either have to plug a custom sink with appropriate assertions, otherwise, I can only think of manual receive with pattern matching both event and buffer and only then asserting the event was first